src/Build/Logging/FancyLogger/FancyLoggerNode.cs

```diff
@@ -4,148 +4,136 @@
+        {
+            Id = args.BuildEventContext!.TargetId;
+            TargetName = args.TargetName;
+        }
+        public void AddTask(TaskStartedEventArgs args)
         {
-            if (Children.ContainsKey(node.Id)) return;
-            Children.Add(node.Id, node);
-            node.Depth = Depth + 1;
-            node.Parent = this;
+            CurrentTaskNode = new FancyLoggerTaskNode(args);
         }
+    }
 
-        public int GetLastLineIndex()
+    public class FancyLoggerTaskNode
+    {
+        public int Id;
+        public string TaskName;
+        public FancyLoggerTaskNode(TaskStartedEventArgs args)
         {
-            // If no line, return -1
-            if (Line == null) return -1;
-            // Get line index and id
-            int lastLineIndex = FancyLoggerBuffer.GetLineIndexById(Line.Id);
-            int lastLineId = Line.Id;
-            if (lastLineIndex == -1) return -1;
-            // Get max of children
-            foreach (var child in Children)
-            {
-                int childLastLineIndex = child.Value.GetLastLineIndex();
-                if (childLastLineIndex > lastLineIndex)
-                {
-                    lastLineIndex = childLastLineIndex;
-                    lastLineId = child.Value.Line!.Id;
-                }
-            }
-            return lastLineIndex;
+            Id = args.BuildEventContext!.TaskId;
+            TaskName = args.TaskName;
         }
+    }
 
-        public void Write()
+    public class FancyLoggerWarningNode
+    {
+        public FancyLoggerWarningNode(BuildWarningEventArgs args)
         {
-            if (Line == null) { return; }
-            // Adjust identation
-            Line.IdentationLevel = Depth - 1;
-            // If line not in the buffer, add
-            if (FancyLoggerBuffer.GetLineIndexById(Line.Id) == -1)
-            {
-                // Get parent last line index
-                if (Parent != null)
-                {
-                    int parentLastLineId = Parent.GetLastLineIndex();
-                    // if (parentLastLineId == -1) throw new Exception("Oops something went wrong");
-                    if (parentLastLineId == -1) return;
-                    // FancyLoggerBuffer.WriteNewLineAfter(Line, parentLastLineId);
-                    FancyLoggerBuffer.WriteNewLineAfterIndex(Line, parentLastLineId);
-                }
-            }
         }
+    }
 
-        public void Collapse()
+    public class FancyLoggerMessageNode
+    {
+        public FancyLoggerMessageNode(BuildMessageEventArgs args)
         {
-            foreach (var child in Children)
-            {
-                if (child.Value.Line == null) continue;
-                FancyLoggerBuffer.HideLine(child.Value.Line.Id);
-                child.Value.Collapse();
-            }
         }
+    }
 
-        public void Expand()
+    public class FancyLoggerErrorNode
+    {
+        public FancyLoggerErrorNode(BuildErrorEventArgs args)
```

Comment:
So does this mean you can't actually see warnings/messages/errors?

Reply:
Nope. Just simplifying and having one class for warnings/messages/errors instead of three. Also allows us to store a list of `FancyLoggerMessageNode` instead of a list of `object`